src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubMessage.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import com.google.pubsub.v1.ReceivedMessage;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.HashMap;
+import javax.annotation.Nonnull;
+
+/**
+ * Spark Receiver for Pub/Sub Messages.
+ * <p>
+ * If backpressure is enabled, the message ingestion rate for this receiver will be managed by Spark.
+ */
+public class PubSubMessage implements Serializable {
```

Comment:
It is actually better to implements `Externalizable` and implements the readExternal / writeExternal methods